### PR TITLE
sdk: grpc::ChannelResponseWriter doesn't need mut

### LINKED ISSUE #598: Make `ChannelResponseWriter` method use non-mut self receiver
It seems these methods could be implemented without needing a `mut` self receiver.

https://github.com/project-oak/oak/blob/d8e0de31d084988528b357ec6418afe389ea27bb/sdk/rust/oak/src/grpc/mod.rs#L67-L71

### PR DESCRIPTION
**Pull request by daviddrysdale**
Fixes #598

# Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [x] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: temp [hunk 1]
```diff
@@ -0,0 +1,80 @@
+diff --git a/Dockerfile b/Dockerfile
```
Comment (rbehjati) [inline temp:1]: Is this file needed to be on git?

Comment (daviddrysdale) [reply temp]: Oops, dinged.
